gpkit/constraints/tight.py

```diff
@@ -17,9 +17,9 @@ def process_result(self, result):
         for constraint in self.flat():
             if isinstance(constraint, (PosynomialInequality,
                                        SignomialInequality)):
-                leftsubbed = constraint.left.sub(variables)
-                rightsubbed = constraint.right.sub(variables)
-                rel_diff = abs(1-(leftsubbed/rightsubbed).value)
+                leftsubbed = constraint.left.sub(variables).value
+                rightsubbed = constraint.right.sub(variables).value
+                rel_diff = abs(1 - leftsubbed/rightsubbed)
```

Comment:
Yeah, I agree. Let's have one process result to generate nus and lambdas,
and another for the rest?
On Mar 25, 2016 10:49, "Warren Hoburg" <notifications@github.com> wrote:

> In gpkit/constraints/tight.py
> <https://github.com/hoburg/gpkit/pull/605#discussion_r57450426>:
>
> > @@ -17,9 +17,9 @@ def process_result(self, result):
> >          for constraint in self.flat():
> >              if isinstance(constraint, (PosynomialInequality,
> >                                         SignomialInequality)):
> > -                leftsubbed = constraint.left.sub(variables)
> > -                rightsubbed = constraint.right.sub(variables)
> > -                rel_diff = abs(1-(leftsubbed/rightsubbed).value)
> > +                leftsubbed = constraint.left.sub(variables).value
> > +                rightsubbed = constraint.right.sub(variables).value
> > +                rel_diff = abs(1 - leftsubbed/rightsubbed)
>
> or do we need separate process_result methods for processing straight out
> of the solver vs processing the packaged result post-check-solution?
>
> I can imagine use cases for both. But usually I think the solution should
> be checked before any processing is done.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/hoburg/gpkit/pull/605/files/985f0f9423d31f3b4f5c32be802336c8b8838667#r57450426>
>